tasks/main.yml

```diff
@@ -44,6 +44,15 @@
     name: "{{ __mssql_server_packages }}"
     state: present
 
+# /opt/mssql/bin/sqlservr requires libldap-2.4.so.2. Latest Fedora use newer
+# libldap by default, therefore, it is required to install openldap-compat
+# because it provides libldap-2.4.so.2.
+- name: Ensure that the openldap-compat package is installed
+  package:
+    name: openldap-compat
+    state: present
+  when: ansible_distribution == "Fedora"
```

Comment:
I think it would be better to add a file `vars/Fedora.yml` like this:
```
__mssql_server_packages: "{{ __mssql_server_packages + ['openldap-compat'] }}"
```
I think that should work - if not, then you'll have to do it like this:
```
__mssql_server_packages: [mssql-server openldap-compat]
```

Reply:
Tests demonstrated that the openldap-compat package is only missing on Fedora 34+, not on Fedora 33.
One way to fix this that I can think of, is to  create `vars/Fedora.yml` with the `openldap-compat` package listed, and `vars/Fedora_33.yml` without that package listed. This will fix the issue for Fedora 33 users, but not for Fedora < 33 users.
Because `vars/` allows configuring variables for separate versions only, not for e.g. Fedora <34, it makes it hard to maintain such a solution.
What you don't like about the task with conditions specifically?
I am adding a condition for ansible_distribution_version too.

Reply:
The issue is that this clutters up tasks/main.yml with a lot of unnecessary platform specific stuff - but ok - I can see that this is a special case - so let's go with this for now, and revisit once we drop support for Fedora 33